sdk/storage/azqueue/queue_client.go

```diff
@@ -0,0 +1,250 @@
+// For more information, see https://learn.microsoft.com/en-us/rest/api/storageservices/set-queue-metadata.
+func (q *QueueClient) SetMetadata(ctx context.Context, options *SetMetadataOptions) (SetMetadataResponse, error) {
+	opts := options.format()
+	resp, err := q.generated().SetMetadata(ctx, opts)
+	return resp, err
+}
+
+// GetProperties gets properties including metadata of a queue.
+// For more information, see https://learn.microsoft.com/en-us/rest/api/storageservices/get-queue-metadata.
+func (q *QueueClient) GetProperties(ctx context.Context, options *GetQueuePropertiesOptions) (GetQueuePropertiesResponse, error) {
+	opts := options.format()
+	resp, err := q.generated().GetProperties(ctx, opts)
+	return resp, err
+}
+
+// GetAccessPolicy returns the queue's access policy.
+// For more information, see https://learn.microsoft.com/en-us/rest/api/storageservices/get-queue-acl.
+func (q *QueueClient) GetAccessPolicy(ctx context.Context, o *GetAccessPolicyOptions) (GetAccessPolicyResponse, error) {
+	options := o.format()
+	resp, err := q.generated().GetAccessPolicy(ctx, options)
+	return resp, err
+}
+
+// SetAccessPolicy sets the queue's permissions.
+// For more information, see https://learn.microsoft.com/en-us/rest/api/storageservices/set-queue-acl.
+func (q *QueueClient) SetAccessPolicy(ctx context.Context, o *SetAccessPolicyOptions) (SetAccessPolicyResponse, error) {
+	opts, acl, err := o.format()
+	if err != nil {
+		return SetAccessPolicyResponse{}, err
+	}
+	resp, err := q.generated().SetAccessPolicy(ctx, acl, opts)
+	return resp, err
+}
+
+// EnqueueMessage adds a message to the queue.
+// For more information, see https://learn.microsoft.com/en-us/rest/api/storageservices/put-message.
+func (q *QueueClient) EnqueueMessage(ctx context.Context, content string, o *EnqueueMessageOptions) (EnqueueMessagesResponse, error) {
+	opts := o.format()
+	message := generated.QueueMessage{MessageText: &content}
+	resp, err := q.messagesClient().Enqueue(ctx, message, opts)
+	return resp, err
+}
+
+// DequeueMessage removes one message from the queue.
+// For more information, see https://learn.microsoft.com/en-us/rest/api/storageservices/get-messages.
+func (q *QueueClient) DequeueMessage(ctx context.Context, o *DequeueMessageOptions) (DequeueMessagesResponse, error) {
+	opts := o.format()
+	resp, err := q.messagesClient().Dequeue(ctx, opts)
+	return resp, err
+}
+
+// UpdateMessage updates a message from the queue with the given popReceipt.
+// For more information, see https://learn.microsoft.com/en-us/rest/api/storageservices/update-message.
+func (q *QueueClient) UpdateMessage(ctx context.Context, messageID string, popReceipt string, content string, o *UpdateMessageOptions) (UpdateMessageResponse, error) {
+	opts := o.format()
+	message := generated.QueueMessage{MessageText: &content}
+	messageURL := runtime.JoinPaths(q.generated().Endpoint(), messageID)
+	messageClient := generated.NewMessageIDClient(messageURL, q.generated().Pipeline())
```

Comment:
Shouldn't this use the internal `MessageIDClient`?

Reply:
that's what I'm doing here, I'm not sure I understand

Reply:
The call to `base.NewQueueClient()` in the client constructor already created a MessageIDClient, so you should be able to use that one instead.

Reply:
ah right, I may be unable to use that since I have to append the message id to the URL, so I'll have to remove the message id client from the composite and keep this code as is

Reply:
@jhendrixMSFT I removed messageID from the composite struct since we need to construct a message specific URL, now we'll just directly construct the generated ID client and perform the operation without exposing messageID client to the customer